src/sage/groups/libgap_mixin.py

```diff
@@ -963,3 +964,166 @@ def is_isomorphic(self, H):
             (False, False, False)
         """
         return self.gap().IsomorphismGroups(H.gap()) != libgap.fail
+
+
+@cached_method
+def minimum_generating_set(group, gap_based=False) -> set:
+    """
+    Return a list containing minimum generating set of the ``group``.
+
+    INPUT:
+
+    - ``group`` -- a group object.
+    - ``gap_based`` -- boolean (default: False). If True, elements of the resulting list are GAP objects.
+
+    OUTPUT:
+
+    A list of elements that generate the group.
+
+    REFERENCE:
+
+    For more information about the algorithm, see https://arxiv.org/abs/2306.07633 or
+    https://www.sciencedirect.com/science/article/pii/S0021869323005720?via%3Dihub.
+
+    EXAMPLES::
+
+        sage: G = GL(2,GF(3))
+        sage: s = G.minimum_generating_set();s
+        [
+        [1 2]  [1 2]
+        [1 1], [0 1]
+        ]
+
+        sage: G = SymmetricGroup(3)
+        sage: s = G.minimum_generating_set(); s
+        [(2,3), (1,2,3)]
+        sage: list(s)[0].parent()
+        Symmetric group of order 3! as a permutation group
+        sage: from sage.groups.libgap_mixin import minimum_generating_set
+        sage: s2 = minimum_generating_set(G, gap_based=True); s2
+        [(2,3), (1,2,3)]
+        sage: list(s2)[0].parent()
+        C library interface to GAP
+
+        sage: A5 = AlternatingGroup(5)
+        sage: A5.minimum_generating_set()
+        [(3,4,5), (1,2,3)]
+
+        sage: H = groups.matrix.Heisenberg(1,3); H
+        Heisenberg group of degree 1 over Ring of integers modulo 3
+        sage: minimum_generating_set(H)
+        [
+        [1 0 0]  [1 1 0]
+        [0 1 1]  [0 1 0]
+        [0 0 1], [0 0 1]
+        ]
+
+    TESTS::
+
+        Check if the elements of the final list are able to generate the group or not::
+
+            sage: p = libgap.eval("DirectProduct(AlternatingGroup(5),AlternatingGroup(5))")
+            sage: s = minimum_generating_set(p,gap_based=True); s
+            [(3,4,5)(8,9,10), (1,2,3,4,5)(6,7,8)]
+            sage: set(p.AsList()) == set(libgap.GroupByGenerators(list(s)).AsList())
+            True
+    """
+    from sage.misc.functional import log
+    from sage.libs.gap.element import GapElement
+
+    if not isinstance(group, GapElement):
+        converter = group._element_constructor_
+        group = group._libgap_()
+    elif not gap_based:
+        return minimum_generating_set(group, gap_based=True)
+
+    if not group.IsFinite().sage():
+        raise NotImplementedError("Implemented for finite group only.")
```

Comment:
```suggestion
        raise NotImplementedError("only implemented for finite groups")
```
This should also have a doctest.